snapshot/snapshot.go

```diff
@@ -119,20 +126,34 @@ func NewSnapshotter(ctx context.Context, cfg *config.Config) (snapshots.Snapshot
 			FsDriver: cfg.FsDriver,
 		})
 		if err != nil {
-			return nil, errors.Wrap(err, "failed to new cache manager")
+			return nil, errors.Wrap(err, "create cache manager")
 		}
 		opts = append(opts, fspkg.WithCacheManager(cacheMgr))
 	}
 
 	// Prefetch mode counts as no daemon, as daemon is only for prefetch,
 	// container rootfs doesn't need daemon
-	hasDaemon := cfg.DaemonMode != config.DaemonModeNone && cfg.DaemonMode != config.DaemonModePrefetch
+	hasDaemon := cfg.DaemonMode != config.DaemonModeNone
 
 	nydusFs, err := fspkg.NewFileSystem(ctx, opts...)
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to initialize nydus filesystem")
 	}
 
+	var blobMgr *blob.Manager
+	if fuseConfig, ok := daemonConfig.(*daemonconfig.FuseDaemonConfig); ok {
+		if fuseConfig.Device.Backend.BackendType == "localfs" &&
```

Comment:
Maybe give a comment to describe what's doing here.

Reply:
Comments added